packages/cashier/src/Containers/cashier.jsx

```diff
@@ -98,15 +99,19 @@ const Cashier = ({
     if ((!is_logged_in && is_logging_in) || !is_account_setting_loaded) {
         return <Loading is_fullscreen />;
     }
+
+    const getHeaderTitle = () => {
+        if (is_default_route && (is_loading || is_cashier_default)) return localize('Cashier');
+
+        return selected_route.getTitle();
+    };
+
     return (
         <FadeWrapper is_visible={is_visible} className='cashier-page-wrapper' keyname='cashier-page-wrapper'>
             <AccountPromptDialog />
             <ErrorDialog />
             <div className='cashier'>
-                <PageOverlay
-                    header={isMobile() && !is_cashier_default ? selected_route.getTitle() : localize('Cashier')}
-                    onClickClose={onClickClose}
-                >
+                <PageOverlay header={isMobile() ? getHeaderTitle() : localize('Cashier')} onClickClose={onClickClose}>
```

Comment:
I think you can combine this logic with the logic inside `getHeaderTitle`

```js
const getHeaderTitle = () => {
    if (
        !isMobile() || 
        (is_default_route && (is_loading || is_cashier_default))
    ) return localize('Cashier');

    return selected_route.getTitle();
};
```

And in the `PageOverlay` simply just call `getHeaderTitle`

```js
<PageOverlay header={getHeaderTitle()} onClickClose={onClickClose}>
```